tests/lang/s02_core/s99_coroutines/t03_macro_cannot_be_coroutine.nim

```diff
@@ -0,0 +1,9 @@
+discard """
+  description: "Macros cannot be coroutines"
+  action: reject
+"""
+
+# XXX: perhaps too restrictive; it's not impossible to implement
+
+macro m() {.coroutine.} =
```

Comment:
I'm trying to think of a use case, but coming up empty. Maybe chunks of AST with diagnostics interleaved? It's sorta what we want in the compiler.

Reply:
I didn't have a specific use-case in mind, my thinking for possibly supporting it was that it's doable and *seems* like it wouldn't do harm. One potential usage:
```nim
var instance {.compileTime.}: Coroutine[NimNode]

macro m2() {.coroutine.} =
  # some other macro

macro m1() {.coroutine.} =
  instance = self
  suspend(launch m2())
  echo "here"

macro finish() {.coroutine.} =
  suspend(move instance)

m1() # invokes `m1`, which suspends with returning a second macro that
     # is then trampolined to completion

finish() # echoes 'here'
```
Whether that's actually useful in practice, I'm not sure yet. In any case, I don't think having this part of the first revision is needed, and adding support for macro coroutines later should not be a problem, implmetation-wise.

Reply:
> Whether that's actually useful in practice, I'm not sure yet. In any case, I don't think having this part of the first revision is needed, and adding support for macro coroutines later should not be a problem, implmetation-wise.

Yup, it makes sense to defer, we might learn about some edge cases where we don't want to do it, or the implemenation will be quite straightforward because the work is done pretty much everywhere else.